net-utils/src/lib.rs

```diff
@@ -102,7 +99,14 @@ fn ip_echo_server_request(
 /// Determine the public IP address of this machine by asking an ip_echo_server at the given
 /// address
 pub fn get_public_ip_addr(ip_echo_server_addr: &SocketAddr) -> Result<IpAddr, String> {
-    ip_echo_server_request(ip_echo_server_addr, IpEchoServerMessage::default())
+    let resp = ip_echo_server_request(ip_echo_server_addr, IpEchoServerMessage::default())?;
+    Ok(resp.address)
+}
+
+pub fn get_cluster_shred_version(ip_echo_server_addr: &SocketAddr) -> Result<u16, String> {
```

Comment:
I think I'd rather roll this into `get_public_ip_addr()` instead of a separate public `get_cluster_shred_version()` function that somebody will 100% use after `get_public_ip_addr()`, causing a 2x load on the entrypoint